### PR TITLE
Add initial support for the Swift Package Manager 

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by MarcelBochtler — APPROVED** [review]
LGTM, I also tested the implementation on my Macbook with swift version `5.8.1`, which worked as expected.

**Comment by sschuberth** [comment]
> LGTM, I also tested the implementation on my Macbook with swift version `5.8.1`, which worked as expected.

I also tested this after-the-fact on Linux with the [Vapor project](https://github.com/vapor/vapor) mentioned [here](https://github.com/oss-review-toolkit/ort/pull/5092#issuecomment-1052778698), and it successfully produced an analyzer result with a good-looking dependency tree.

File: plugins/package-managers/spm/src/funTest/assets/projects/synthetic/spm-app/Package.resolved [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (fviernau) [inline plugins/package-managers/spm/src/funTest/assets/projects/synthetic/spm-app/Package.resolved:1]: @UgniusV  I'm trying to enhance the implementation and I'm surprised by the entries in this lockfile.
Can you please explain how this file has been created? 

In particular, has the file been generated by the `swift` command? Has it been modified or been crafted by hand?
(I'm after the combination of `branch` `revision` `version`. E.g. which ones can be present. After playing around a while with `spm` I was unable to make it generate anything like this.

Comment (fviernau) [reply plugins/package-managers/spm/src/funTest/assets/projects/synthetic/spm-app/Package.resolved:1]: If you could provide instructions how to reproduce the creation steps of this file...

**Pull request merged**